EntranceRando.py

```diff
@@ -298,10 +308,10 @@ def find_pairing(dead_end: bool, require_new_regions: bool) -> Optional[Tuple[En
             for target_entrance in entrance_lookup.get_targets(target_groups, dead_end, preserve_group_order):
                 # TODO - requiring new regions is a proxy for requiring new entrances to be unlocked, which is
                 #        not quite full fidelity so we may need to revisit this in the future
-                region_requirement_satisfied = (not require_new_regions
-                                                or target_entrance.connected_region not in er_state.placed_regions)
-                if region_requirement_satisfied and source_exit.can_connect_to(target_entrance, er_state):
-                    return source_exit, target_entrance
+                if ((not require_new_regions or target_entrance.connected_region not in er_state.placed_regions)
```

Comment:
I quite dislike this change, breaking down complex boolean conditions in cases such as this can really help readability. Not sure what is more likely to ultimately get merged though

Reply:
yeah i think this one comes down to preference so i can undo it